corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -553,11 +549,12 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
 
         _updateModelsForValidation: function () {
             var self = this;
+            var promise = $.Deferred();
+            self.updateModelsForValidation = promise;
```

Comment:
Since it's just a reference to the promise object, I should be able to reliably check the state of that promise at any point after the promise is created. Before it's created, `validateAllFields` will look it up, find `undefined` and just run this function itself. Once it's resolved it should stay resolved until/unless it's called again. The idea is if the function is already running (or completed), don't run it again.

I think it may theoretically be possible to check the state between a field changing and `self.updateModelsForValidation` getting set, maybe finding it resolved from the previous iteration rather than pending, but in practice I don't know of a way that would ever happen. Does that sort of sort answer your question?

Reply:
Yes that makes sense, thanks for the explanation.